### PR TITLE
add updated project header buttons

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

Updates the project header social links on project page to now match figma, and use tailwind where possible.

tested on:
- dark/light mode
- mobile / desktop
- v1 project and v2/3 project

When testing this PR it's worth going through the deploy preview and testing on all 6 test cases.

- Other notes
- some colors are hard coded via tail wind. I did a quick search through the project and `tailwind.config.js` but the color wasn't defined, so I reverted to the `[#fffffff]` hex color syntax.

One small bug I was having an issue with. The color of the buttons isn't changing on the toggle between light and dart. But, is correct on hard reload. I think it might be a thing with tailwind. I didn't want to spin my wheels on it too long, I figured someone here would prob know.

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

https://www.figma.com/file/SccJrdCJYOvKY7ObcK99Nk/JBX---Project-Page---Social-Icons-Update---Sep-2022?node-id=152%3A22212&t=KqEVFGOB8KxcpcOD-1
## Screenshots or screen recordings

Quick screen shot for context: 
image

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/icons/Discord.tsx [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
...48.4172 21.9935 48.2735 21.8676 48.2256C19.9366 47.4931 18.0979 46.6 16.3292 45.5858C16.1893 45.5041 16.1781 45.304 16.3068 45.2082C16.679 44.9293 17.0513 44.6391 17.4067 44.3461C17.471 44.2926 17.5606 44.2813 17.6362 44.3151C29.2558 49.6202 41.8354 49.6202 53.3179 44.3151C53.3935 44.2785 53.4831 44.2898 53.5502 44.3433C53.9057 44.6363 54.2779 44.9293 54.6529 45.2082C54.7816 45.304 54.7732 45.5041 54.6333 45.5858C52.8646 46.6197 51.0259 47.4931 49.0921 48.2228C48.9662 48.2707 48.9102 48.4172 48.9718 48.5383C50.038 50.6034 51.2554 52.5699 52.5959 54.435C52.6519 54.5139 52.7526 54.5477 52.845 54.5195C58.6464 52.7249 64.529 50.0174 70.6019 45.5576C70.6551 45.5182 70.6887 45.459 70.6943 45.3942C72.1747 30.0791 68.2147 16.7757 60.1968 4.9823C60.1772 4.9429 60.1437 4.9147 60.1045 4.8978ZM23.7259 37.3253C20.2276 37.3253 17.3451 34.1136 17.3451 30.1693C17.3451 26.225 20.1717 23.0133 23.7259 23.0133C27.308 23.0133 30.1626 26.2532 30.1066 30.1693C30.1066 34.1136 27.28 37.3253 23.7259 37.3253ZM47.3178 37.3253C43.8196 37.3253 40.9371 34.1136 40.9371 30.1693C40.9371 26.225 43.7636 23.0133 47.3178 23.0133C50.9 23.0133 53.7545 26.2532 53.6986 30.1693C53.6986 34.1136 50.9 37.3253 47.3178 37.3253Z"
```
Comment (benschac) [inline src/components/icons/Discord.tsx:23]: I thought this was my prettier, guessing not

File: src/components/icons/Telegram.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      viewBox="0 0 17 15"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path
+        id="telegram-1"
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M13.022 14.0102C13.2501 14.1717 13.544 14.2121 13.8061 14.113C14.0682 14.0131 14.2608 13.7892 14.3189 13.518C14.9344 10.6251 16.4276 3.30307 16.9879 0.671606C17.0304 0.473272 16.9595 0.267147 16.8037 0.134689C16.6479 0.00223004 16.4318 -0.0360201 16.2392 0.0355217C13.2692 1.13486 4.12269 4.56673 0.38419 5.95011C0.146903 6.03794 -0.00750968 6.26603 0.000281807 6.51607C0.00878161 6.76682 0.177361 6.98428 0.420314 7.05724C2.0969 7.55874 4.29764 8.25645 4.29764 8.25645C4.29764 8.25645 5.32612 11.3625 5.86231 12.9421C5.9296 13.1404 6.08473 13.2962 6.28943 13.3501C6.49343 13.4032 6.71159 13.3472 6.86388 13.2035C7.72519 12.3903 9.05683 11.133 9.05683 11.133C9.05683 11.133 11.5869 12.9881 13.022 14.0102ZM5.22341 7.86403L6.41268 11.7868L6.67688 9.30266C6.67688 9.30266 11.2717 5.15819 13.8911 2.7959C13.9676 2.72648 13.9782 2.61032 13.9145 2.52886C13.8514 2.4474 13.7353 2.42827 13.6481 2.48352C10.6123 4.42224 5.22341 7.86403 5.22341 7.86403Z"
+        fill={color}
```
Comment (benschac) [inline src/components/icons/Telegram.tsx:27]: Not sure why this svg export is acting differently, but, this was the only way I could get the svg fill to work

File: src/components/icons/Discord.tsx (hunk 1 above)
Comment (benschac) [inline src/components/icons/Discord.tsx:23]: i think this `m` is uppercase, will come back and fix

File: src/components/Project/ProjectHeader/SocialLinks.tsx [hunk 3]
```diff
@@ -1,51 +1,87 @@
...
   infoUri?: string
   twitter?: string
   discord?: string
+  telegram?: string
 }) {
+  const isMobile = useMobile()
+  const { themeOption } = useJuiceTheme()
+
+  const styleObj = {
+    fontSize: isMobile ? 22 : 16,
```
Comment (benschac) [inline src/components/Project/ProjectHeader/SocialLinks.tsx:47]: needed to use a style object to apply it to icon / SVG.

File: src/components/Project/ProjectHeader/SocialLinks.tsx [hunk 4]
```diff
@@ -1,51 +1,87 @@
...
+import { useJuiceTheme } from 'hooks/JuiceTheme'
+import useMobile from 'hooks/Mobile'
+import colors from 'tailwindcss/colors'
+
+type SocialProps = {
+  children: React.ReactNode
+  link: string
+}
+
+function SocialButton(props: SocialProps) {
```
Comment (benschac) [inline src/components/Project/ProjectHeader/SocialLinks.tsx:15]: these buttons are just used in this one file and thought co-locating the button component was easier to grok.

**Review by tomquirk — APPROVED** [review]

**Comment by tomquirk** [comment]
Thanks for starting this off @benschac! I made a few changes and will merge when pipeline succeeds ✅

**Pull request merged**